# 📊 Parcial – Análisis de Redes Sociales (Primer Corte)

## 🎯 Objetivo
Evaluar tu dominio de **filtros** y **agrupaciones** en Python (sin pivots ni gráficos) y tu capacidad de **análisis de negocio** en métricas de marketing: **CAC, LTV, churn rate, funnel, unit economics (LTV/CAC)**.

---
## 🧩 Contexto de Negocio (Caso Real)
**StartUp SaaS en crecimiento.** Vende planes **Basic, Pro, Enterprise** en **LatAm, North America, Europe, APAC**. Los canales de adquisición son: **meta_ads, google_ads, organic_search, outbound_sales**. Los usuarios se registran por **web** o **mobile_app**.

La dirección quiere decidir **dónde invertir el presupuesto del próximo trimestre**. Te piden:

1) Identificar **canales más rentables** (relación **LTV/CAC** y churn).  
2) Detectar **regiones** con mayor **potencial de crecimiento** y/o **riesgo**.  
3) Evaluar el desempeño por **tier** (Basic, Pro, Enterprise).  
4) Recomendar **asignación de presupuesto** por **canal + tier**.

---
## 📂 Instrucciones
- Trabaja únicamente con **filtros** y **agrupaciones**. **No uses** tablas dinámicas (pivot), merges ni gráficos.
- Archivo a usar: `clientes_marketing.csv`.
- Escribe **código + interpretación breve** para cada inciso.
- Si el enunciado dice **“Agrupa”**, usa `groupby`. Si dice **“Filtra”**, usa filtrado con máscaras booleanas. Si dice **“Ambas”**, realiza primero el **filtro** y luego la **agrupación**.

---


## Descargar Tabla (Correr una vez)

In [1]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv

--2025-09-17 23:22:44--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv [following]
--2025-09-17 23:22:44--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 661398 (646K) [text/plain]
Saving to: ‘cac_ltv_model.csv’

cac_ltv_model.csv   100%[===================>] 645.90K  --.-KB/s

## Importar Datos

In [3]:
import pandas as pd
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')

# 1) Cargar dataset
data = pd.read_csv('cac_ltv_model.csv')


### ✅ Variables relevantes (recordatorio)
- `acquisition_channel`, `signup_source`, `region`, `customer_tier`
- `plan_price`, `discount_rate`, `arpu`, `gross_margin`, `churn_rate`
- `contract_length_months`, `marketing_spend`

### 🧮 Fórmulas de negocio (a usar en el examen)
- **CAC por canal**:  
    $$ CAC = \frac{\text{Total Marketing Spend}}{\text{Clientes adquiridos}} $$
  
- **LTV por observación** (aprox.):  
   $$ LTV = ARPU \times contract\_length\_months \times gross\_margin \times (1 - churn\_rate) $$

- **Unit economics**:  
  $$ LTV/CAC   $$(ideal \> 3 como regla general de SaaS)

---



### Parte Exploratoria: Cree una muestra de los primeros 5 elementos de esta tabla. ¿Cuantas observaciones tienes?

In [7]:
data.head(5)

,year,month,date,customer_id,acquisition_channel,signup_source,region,customer_tier,plan_price,discount_rate,arpu,gross_margin,churn_rate,contract_length_months,marketing_spend
0,2023,1,Jan-23,1001,outbound_sales,web,North America,Basic,78.84,0.10,63.63,0.76,0.02,12,212.48
1,2023,1,Jan-23,1002,meta_ads,web,LatAm,Pro,120.23,0.00,124.47,0.84,0.05,1,352.60
2,2023,1,Jan-23,1003,organic_search,mobile_app,North America,Enterprise,335.82,0.20,278.20,0.81,0.02,12,55.40
3,2023,1,Jan-23,1004,organic_search,web,Europe,Pro,193.89,0.10,190.65,0.80,0.08,1,49.10
4,2023,1,Jan-23,1005,organic_search,web,APAC,Enterprise,471.80,0.10,445.32,0.80,0.08,1,45.27


In [8]:
print("shape (filas, columnas):", data.shape)

shape (filas, columnas): (7057, 15)


## 🔎 Parte A – Filtros

hint: Crea una tabla con el filtro y luego agrupa la variable que te piden.

1. **Filtra** los clientes del canal **`meta_ads`**. Calcula el **ARPU promedio** de este subconjunto.  



In [13]:
meta_ads = data[data['acquisition_channel'] == 'meta_ads']

In [12]:
arpu_meta = meta_ads['arpu'].mean()
print("ARPU promedio meta_ads:", arpu_meta)

ARPU promedio meta_ads: 168.78858272162617


In [15]:
meta_ads.head(5)

,year,month,date,customer_id,acquisition_channel,signup_source,region,customer_tier,plan_price,discount_rate,arpu,gross_margin,churn_rate,contract_length_months,marketing_spend
1,2023,1,Jan-23,1002,meta_ads,web,LatAm,Pro,120.23,0.00,124.47,0.84,0.05,1,352.60
10,2023,1,Jan-23,1011,meta_ads,web,North America,Enterprise,341.61,0.00,342.49,0.89,0.07,1,376.83
15,2023,1,Jan-23,1016,meta_ads,mobile_app,Europe,Enterprise,282.59,0.00,293.98,0.85,0.07,1,355.55
19,2023,1,Jan-23,1020,meta_ads,mobile_app,North America,Pro,197.22,0.00,181.95,0.76,0.03,12,363.98
23,2023,1,Jan-23,1024,meta_ads,mobile_app,Middle East,Pro,138.31,0.10,139.64,0.82,0.08,1,367.74


2. **Filtra** clientes de **LatAm** con **`churn_rate > 0.05`**. ¿Cuántos son? ¿Cuál es su **ARPU promedio**?  

In [19]:
latam_churn = data[(data['region'] == 'LatAm') & (data['churn_rate'] > 0.05)]
arpu_prom = meta_ads['arpu'].mean()
print("arpu prom (meta_ads):", arpu_prom)

arpu prom (meta_ads): 168.78858272162617


In [20]:
latam_churn.head(5)

,year,month,date,customer_id,acquisition_channel,signup_source,region,customer_tier,plan_price,discount_rate,arpu,gross_margin,churn_rate,contract_length_months,marketing_spend
1,2023,1,Jan-23,1002,meta_ads,web,LatAm,Pro,120.23,0.00,124.47,0.84,0.05,1,352.60
25,2023,1,Jan-23,1026,outbound_sales,web,LatAm,Basic,70.41,0.00,71.94,0.90,0.07,1,181.45
49,2023,1,Jan-23,1050,meta_ads,web,LatAm,Basic,65.13,0.10,66.08,0.89,0.08,1,305.42
65,2023,1,Jan-23,1066,organic_search,partner,LatAm,Pro,160.83,0.20,129.87,0.86,0.06,1,44.50
92,2023,1,Jan-23,1093,google_ads,partner,LatAm,Pro,185.70,0.00,188.58,0.79,0.08,1,411.41


3. **Filtra** clientes **Enterprise** con `contract_length_months > 6`. ¿Cuál es su **gross_margin promedio**?

In [22]:
enterprise_contract = data[(data['customer_tier'] == 'Enterprise') & (data['contract_length_months'] > 6)]
gross_margin_prom = enterprise_contract['gross_margin'].mean()
print("gross_margin prom (Enterprise):", gross_margin_prom)

gross_margin prom (Enterprise): 0.8248397435897435


## 📊 Parte B – Agrupaciones (SOLO agrupar)
4. **Agrupa** por `acquisition_channel` y calcula **ARPU promedio**. Ordena de mayor a menor.  


In [33]:
arpu_prom_channel = data.groupby('acquisition_channel')['arpu'].mean().sort_values(ascending=False)
print(arpu_prom_channel)

acquisition_channel
outbound_sales   173.61
google_ads       172.37
organic_search   171.14
meta_ads         168.79
Name: arpu, dtype: float64


5. **Agrupa** por `region` y calcula **churn_rate promedio**. Identifica la región con mayor churn.  



In [35]:
churn_rate_prom_region = data.groupby('region')['churn_rate'].mean().sort_values(ascending=False)
print("churn_rate promedio por region:")
print(churn_rate_prom_region)

churn_rate promedio por region:
region
Middle East     0.05
LatAm           0.05
North America   0.05
APAC            0.05
Europe          0.05
Africa          0.05
Name: churn_rate, dtype: float64


6. **Agrupa** por `customer_tier` y calcula **marketing_spend total**. ¿Cuál tier consume más presupuesto?

In [32]:
marketing_spend_total = data.groupby('customer_tier')['marketing_spend'].sum()
print("marketing_spend total por tier:")
print(marketing_spend_total)

marketing_spend total por tier:
customer_tier
Basic        584,419.11
Enterprise   584,348.87
Pro          589,265.04
Name: marketing_spend, dtype: float64


## 🔀 Parte C – Filtro **y** Agrupación (Ambas)
7. **Filtra** solo registros de **`signup_source = 'web'`** y luego **agrupa** por `acquisition_channel` para obtener el **churn_rate promedio**.  


In [41]:
web_users = data[data['signup_source'] == 'web']
churn_web_channel = web_users.groupby('acquisition_channel')['churn_rate'].mean()

print("Churn promedio por canal")
print(churn_web_channel)

Churn promedio por canal
acquisition_channel
google_ads       0.05
meta_ads         0.05
organic_search   0.05
outbound_sales   0.05
Name: churn_rate, dtype: float64


8. **Filtra** solo **`mobile_app`** y luego **agrupa** por `region` para calcular **ARPU promedio**.  


In [40]:
mobile_app_users = data[data['signup_source'] == 'mobile_app']
arpu_mobile_region = mobile_app_users.groupby('region')['arpu'].mean().sort_values(ascending=False)
print("ARPU promedio por region")
print(arpu_mobile_region)

ARPU promedio por region
region
North America   174.00
LatAm           173.20
Africa          172.60
Middle East     171.71
APAC            169.09
Europe          165.89
Name: arpu, dtype: float64


9. **Filtra** a clientes **`Pro`** y **agrupa** por `acquisition_channel` para calcular **marketing_spend total**.


In [42]:
acquis_channel_pro = data[data['customer_tier'] == 'Pro']
marketing_spend_pro = acquis_channel_pro.groupby('acquisition_channel')['marketing_spend'].sum()
print("marketing_spend total por canal")
print(marketing_spend_pro)

marketing_spend total por canal
acquisition_channel
google_ads       226,721.68
meta_ads         211,517.60
organic_search    29,848.72
outbound_sales   121,177.04
Name: marketing_spend, dtype: float64


## 📈 Parte D – Métricas de negocio (CAC, LTV, LTV/CAC)
10. **CAC por canal (Agrupa)**: calcula el CAC de cada `acquisition_channel` como:  
   `CAC = marketing_spend_total_del_canal / #clientes_del_canal`  


In [45]:
cac_channel = (data.groupby('acquisition_channel')['marketing_spend'].sum() / data.groupby('acquisition_channel').size())
print("CAC por canal:")
print(cac_channel.sort_values(ascending=False))

CAC por canal:
acquisition_channel
google_ads       401.02
meta_ads         349.27
outbound_sales   200.22
organic_search    49.95
dtype: float64


11. **CAC por tier (Agrupa)**: calcula el CAC de cada `customer_tier` como:  
   `CAC = marketing_spend_total_del_tier / #clientes_del_canal`  


In [46]:
cac_tier = (data.groupby('customer_tier')['marketing_spend'].sum() / data.groupby('customer_tier').size())
print("CAC por tier:")
print(cac_tier.sort_values(ascending=False))

CAC por tier:
customer_tier
Basic        249.75
Enterprise   249.19
Pro          248.43
dtype: float64


12. **LTV por tier (Ambas)**: crea una columna `ltv_individual` con la fórmula dada.


In [54]:
ltv_individual = (data['arpu'] * data['contract_length_months'] * data['gross_margin'] * (1 - data['churn_rate']))
data['ltv_individual'] = ltv_individual
data.head(5)

,year,month,date,customer_id,acquisition_channel,signup_source,region,customer_tier,plan_price,discount_rate,arpu,gross_margin,churn_rate,contract_length_months,marketing_spend,ltv_individual
0,2023,1,Jan-23,1001,outbound_sales,web,North America,Basic,78.84,0.10,63.63,0.76,0.02,12,212.48,567.54
1,2023,1,Jan-23,1002,meta_ads,web,LatAm,Pro,120.23,0.00,124.47,0.84,0.05,1,352.60,99.20
2,2023,1,Jan-23,1003,organic_search,mobile_app,North America,Enterprise,335.82,0.20,278.20,0.81,0.02,12,55.40,"2,651.91"
3,2023,1,Jan-23,1004,organic_search,web,Europe,Pro,193.89,0.10,190.65,0.80,0.08,1,49.10,139.68
4,2023,1,Jan-23,1005,organic_search,web,APAC,Enterprise,471.80,0.10,445.32,0.80,0.08,1,45.27,328.72


12. A. Luego **agrupa** `ltv_individual` por
`acquisition_channel` para obtener el **LTV promedio**.  

In [53]:
groupby_channel = data.groupby('acquisition_channel')['ltv_individual'].mean().sort_values(ascending=False)
print("LTV promedio por acquisition_channel:")
print(groupby_channel)

LTV promedio por acquisition_channel:
acquisition_channel
organic_search   765.13
outbound_sales   751.57
meta_ads         729.05
google_ads       710.25
Name: ltv_individual, dtype: float64


12. B. Luego **agrupa** `ltv_individual`por `customer_tier` para obtener el **LTV promedio**.





In [57]:
LTV = data.groupby('customer_tier')['ltv_individual'].mean().sort_values(ascending=False)

print("LTV promedio por canal:")
print(LTV)

LTV promedio por canal:
customer_tier
Enterprise   1,349.02
Pro            577.63
Basic          291.77
Name: ltv_individual, dtype: float64


13. **Unit economics (Ambas)**: combina tus resultados para comparar **LTV promedio por tier** contra **CAC por canal** y comenta **qué combinaciones canal + tier** lucen más saludables (busca **LTV/CAC > 3**).


In [61]:
LTV = data.groupby(['acquisition_channel'])['ltv_individual'].mean().sort_values(ascending=False)
LTV

,ltv_individual
acquisition_channel,
organic_search,765.13
outbound_sales,751.57
meta_ads,729.05
google_ads,710.25


In [66]:
CAC = data.groupby(['acquisition_channel'])['marketing_spend'].sum() / data.groupby('acquisition_channel')['customer_id'].count()
CAC = CAC.sort_values(ascending=False)
CAC

,0
acquisition_channel,
google_ads,401.02
meta_ads,349.27
outbound_sales,200.22
organic_search,49.95


In [69]:
UNIT_ECONOMICS = LTV / CAC
print(UNIT_ECONOMICS.sort_values(ascending=False))

acquisition_channel
organic_search   15.32
outbound_sales    3.75
meta_ads          2.09
google_ads        1.77
dtype: float64


In [70]:
LTV = data.groupby(['customer_tier'])['ltv_individual'].mean().sort_values(ascending=False)
LTV

,ltv_individual
customer_tier,
Enterprise,"1,349.02"
Pro,577.63
Basic,291.77


In [71]:
CAC = data.groupby(['customer_tier'])['marketing_spend'].sum() / data.groupby('customer_tier')['customer_id'].count()
CAC = CAC.sort_values(ascending=False)
CAC

,0
customer_tier,
Basic,249.75
Enterprise,249.19
Pro,248.43


In [72]:
UNIT_ECONOMICS = LTV / CAC
print(UNIT_ECONOMICS.sort_values(ascending=False))

customer_tier
Enterprise   5.41
Pro          2.33
Basic        1.17
dtype: float64


In [77]:
import pandas as pd

# 1) Definir los LTV promedio por tier (de tus cálculos anteriores)
ltv_tier = {
    "Enterprise": 1349.02,
    "Pro": 577.63,
    "Basic": 291.77
}

# 2) Definir los CAC por canal (de tus cálculos anteriores)
cac_channel = {
    "google_ads": 401.02,
    "meta_ads": 349.27,
    "outbound_sales": 200.22,
    "organic_search": 49.95
}

# 3) Convertirlos a DataFrames
ltv_df = pd.DataFrame.from_dict(ltv_tier, orient="index", columns=["LTV"])
cac_df = pd.DataFrame.from_dict(cac_channel, orient="index", columns=["CAC"])

# 4) Calcular la matriz de LTV/CAC (cruce tier × canal)
results = pd.DataFrame(
    {
        canal: ltv_df["LTV"] / cac
        for canal, cac in cac_channel.items()
    }
)

# 5) Redondear a 2 decimales para visualización
results = results.round(2)

print("Matriz LTV/CAC (Tier x Canal):")
print(results)


Matriz LTV/CAC (Tier x Canal):
            google_ads  meta_ads  outbound_sales  organic_search
Enterprise        3.36      3.86            6.74           27.01
Pro               1.44      1.65            2.88           11.56
Basic             0.73      0.84            1.46            5.84


Lucen más saludables:
* Organic search/outbound sales + enterprice

## 🧠 Parte E – Análisis crítico (respuesta abierta)
13. **CMO por un día**: con tus resultados, ¿en qué **canal(es)** invertirías más el próximo trimestre? ¿Por qué? (Cita **CAC** y **churn**).  
14. ¿Qué **región** ves con mayor riesgo? ¿Qué hipótesis explicarían su **churn**?  
15. Identifica **dos combinaciones canal + tier** con mejor **LTV/CAC** y propón **dos acciones** concretas de optimización (p. ej., creatividades, audiencias, pricing, retención).

---
✍️ **Nota**: Mantente disciplinado con el enunciado (**Filtra**, **Agrupa**, **Ambas**). La evaluación pondera **correctitud técnica** y **calidad de interpretación de negocio**.


**13.**
En el próximo trimestre invertiría más en Organic search ya que es el más eficiente teniendo en cuenta que el CAC es bajo (49.95 USD); en outbound  sales al tener un CAC alto (220.22 USD) tiene un LTV aceptable (3.75) y en Meta Ads, unicamente porque tiene mayor escalabilidad pero su CAC es muy más alto (349.27 USD) y sería solo en enterprice.

**14.**
Todas las regiones presentan el mismo churn promedio (5%) y el ARPU más bajo esta en Europa. Sin embargo, América [Norte (174.00) y LatAm (173.20)] tiene ARPU móviles cercanos y por esto se puede decir que en LatAm puede haber un riesgo mayor debido a la sensibilidad de precios. Sería bueno que se realicen cortes mensuales por región que ayuden a evaluar mejor el comportamiento.

**15.**
1. Organic search + enterprice = 27.01(LTV/CAC) => Muy rentable

Acción 1: Potenciar adquisión con SEO y contenido B2B por medio de keywords y estrategia de linkbuilding.
Acción 2: Incentivar las metricas de ROI e incentivar contratos anuales con descuentos.

2. Organic Search + Pro = 11.56 (LTV/CAC) => rentable

Acción 1: Optimizar la conversión tráfico a pago/compra por medio de automatización en el onboarding de la App/Web y ofrecer promociones ligadas a las conversiones de pago.
Acción 2: Retención y fidelización por medio de upgrades en servicio a menor costo. Tambíen se deberá hacer mayor seguimiento al detectar señales de churn (inactividad, no pago).